isisd/isis_tlvs.c

```diff
@@ -3480,14 +3484,16 @@ static void format_tlv_router_cap(const struct isis_router_cap *router_cap,
 			  router_cap->srlb.range_size);
 
 	/* Segment Routing Algorithms as per RFC8667 section #3.2 */
-	if (router_cap->algo[0] != SR_ALGORITHM_UNSET) {
+	nb_algo = 0;
+	for (int i= 0; i < SR_ALGORITHM_COUNT; i++)
+		if (router_cap->algo[i])
```

Comment:
warning with sbuf size limited to 4096. may be not enough if the range is increased.